hibernate-core/src/test/java/org/hibernate/orm/test/annotations/collectionelement/QueryTest.java

```diff
@@ -21,14 +27,38 @@
 @SessionFactory
 public class QueryTest {
 
+	@BeforeEach
+	public void setUp(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					EntityWithAnElementCollection entityWithAnElementCollection = new EntityWithAnElementCollection();
+					entityWithAnElementCollection.setId( 1L );
+					entityWithAnElementCollection.addSomeString( "abc" );
+					entityWithAnElementCollection.addSomeString( "efg" );
+
+					session.save( entityWithAnElementCollection );
+
+					EntityWithAnElementCollection entityWithAnElementCollection2 = new EntityWithAnElementCollection();
+					entityWithAnElementCollection2.setId( 2L );
+					entityWithAnElementCollection2.addSomeString( "hil" );
+					entityWithAnElementCollection2.addSomeString( "mnp" );
+
+					session.save( entityWithAnElementCollection2 );
+				}
+		);
+	}
+
 	@Test
 	@TestForIssue(jiraKey = "HHH-5209")
 	public void testMemberOfSyntax(SessionFactoryScope scope) {
 		// performs syntax checking of the MEMBER OF predicate against a basic collection
 		scope.inSession(
-				session ->
-						session.createQuery( "from EntityWithAnElementCollection e where 'abc' member of e.someStrings" )
-								.list()
+				session -> {
+					List<EntityWithAnElementCollection> list = session.createQuery(
+							"from EntityWithAnElementCollection e where 'abc' member of e.someStrings" )
+							.list();
+					assertThat( list.size(), is( 1 ) );
+				}
 		);
```

Comment:
I would suggest a testing case in which `e.someStrings` is empty to see whether exception is thrown (because of the erroneous `in ()` SQL which only works in h2).

Reply:
This particular one, no.  For a member-of predicate it would be illegal for the sub-query to be empty.

However, you have a great point about other cases for in-predicates in the source query.  I'm not sure `in(null)` works on all databases - seems we have had this discussion before a few years ago.

Reply:
You mean that `someStrings` for a particular `e` has no elements (e.getSomeStrings().isEmpty() == true)?  That's still ok.  The sub-query we generate would simply return no rows.  But the in-predicate sub-query expression itself is not null.  Meaning, in SQL, this is perfectly valid whether the sub-query returns no rows or more than zero:

```
select e 
from entity_with_elem_coll e
where 'abc' in (
    select ...
)
where i
```